combat-trainer.lic

```diff
@@ -2351,8 +2351,8 @@ class TrainerProcess
       'App Quick' => 'Appraisal',
       'App' => 'Appraisal',
       'Astro' => 'Astrology',
+      'Barb Research' => 'Magic', # Auto-select magic skill to train
       'Barb Research Augmentation' => 'Augmentation',
```

Comment:
Do we still need the old researches? I.e. does this new one encompass the rest?

Reply:
Good question. Although technically if the player had "Barb Research Augmentation: 200" in their yaml then this code change would handle it, but hat training ability would run regardless if the Augmentation skill needed to be trained then or not because the word "Magic" isn't a skill so it's learning rate will be 0 and that will always be less than the training abilities target rate.

If we keep "Barb Research <skill>" => "<skill>" then if a player were to be using those specific variants then combat-trainer would only attempt the research if that skill needed to be trained, otherwise it'd pick a different training ability to do.

Because of that nuance in choosing which training ability to perform, I chose to leave it in case someone really wanted that behavior.

Reply:
Does that mean `Barb Research` (no specified skill) will always run, even if all skills are above the threshold?

Reply:
> Does that mean `Barb Research` (no specified skill) will always run, even if all skills are above the threshold?

Hrmm... yes, yes it would... 🤔 

**Unless** the `check_ability` method gets a bit smarter to let us specify one or more skills in that map

Thoughts on doing this differently?

Reply:
Actually... looking at `check_ability`... I think it _might_ support multiple skills already, stand by, will test!

Reply:
Thanks for the questions, I've updated `Barb Research` to specify the three magic skills it's relevant for, and `check_ability` will skip that if those three skills don't need training 👍 

Resolved in https://github.com/rpherbig/dr-scripts/pull/5775/commits/96171ece29d0b04bf4894dd061e076d305d19ca7